packages/discovery/src/discovery.service.ts

```diff
@@ -223,7 +223,7 @@ export class DiscoveryService {
     return {
       name: wrapper.name as string,
       instance: instanceHost.instance,
-      injectType: wrapper.metatype,
+      injectType: wrapper.metatype ?? undefined,
```

Comment:
@xballoy Shall we add a todo here so that when V10 is removed, we can revert it to the expected `null` in the future?
Backfill is fine for now as we're going to support both v10 and v11 but if devs fully adopt v11, this could be a library-specific behavior

Reply:
Yes, we could add a reminder here for if we drop NestJS 10 support.

But at the same time, I guess we could change `DiscoveredClass.injectType` to be able to be null, I think than most of the time people are handling null/undefined the same way so this wouldn't really be a breaking change (maybe on the types). I just wanted the change to be minimal as I wasn't sure about the backward compatibility you wanted.

I'm open to both solutions, just tell me what you prefer.

Reply:
I agree with you, we can add the reminder and also improve the injectType.
As long as we're clear about that breaking change, we're good. I'd like to prepare devs that we're working towards the Nestjs norm on that front.

Reply:
I changed the following:
- keep returning undefined instead of null but adding a TODO for when we'll remove NestJS 10
- I added null to `DiscoveredClass.injectType` types so we prepare support for NestJS 11 (also added a TODO to remove the undefined)